BALSAMIC/assets/scripts/filter_SVs.py

```diff
@@ -0,0 +1,350 @@
+    sv_id_name = "_".join(sv_id_split[0:2])
+    sv_id_num = sv_id_split[2]
+
+    # define unique breakend_name
+    region_a = str(info["REGIONA"][0]) + "_" + str(info["REGIONA"][1])
+    bnd_id = f"{sv_id_name}_{region_a}"
+    return bnd_id, sv_id_num
+
+
+def rescue_t_n_mixed_bnds(vcf_path):
+    """
+    Removes soft filters and sets PASS to BND variants with mixed soft / PASS filters set.
+    Annotates BND variants with unique ID.
+
+    :param vcf_path: Path to VCF
+
+    Outputs to standard out.
+    """
+    vcf_start = vcfpy.Reader.from_path(vcf_path)
+
+    # First read of VCF-file:
+    # define dict with bnd_id - bnd_num - FILTER
+    bnd_filter_dict = {}
+    for sv in vcf_start:
+        info = dict([*sv.INFO.items()])
+
+        svtype = info["SVTYPE"]
+        # only relevant for bnd-variants
+        if svtype != "BND":
+            continue
+        # no need to rescue normal variants
+        if "TUMOR_PASS_CHROM" not in info:
+            continue
+
+        bnd_id, sv_id_num = get_bnd_id(info)
+
+        if bnd_id not in bnd_filter_dict:
+            bnd_filter_dict[bnd_id] = {}
+
+        if sv_id_num in bnd_filter_dict[bnd_id]:
+            logging.warning(
+                f"Conflicting BND-names: {bnd_id}, will not attempt to rescue: {sv}"
+            )
+            continue
+
+        bnd_filter_dict[bnd_id][sv_id_num] = sv.FILTER
+
+    # Second read fo VCF-file:
+    # Annotate and remove BND filters with PASS
+    vcf = vcfpy.Reader.from_path(vcf_path)
+    vcf.header.add_info_line(
+        vcfpy.OrderedDict(
+            [
+                ("ID", "BND_ID"),
+                ("Type", "String"),
+                ("Description", "Unique ID for BND-variants, SVID + REGIONA"),
+            ]
+        )
+    )
+    vcf.header.add_filter_line(
+        vcfpy.OrderedDict(
+            [
+                ("ID", "PASS"),
+                (
+                    "Description",
+                    "If at least 1 of the 2 BND_ID variants have PASS, set PASS",
+                ),
+            ]
+        )
+    )
+
+    writer = vcfpy.Writer.from_path("/dev/stdout", vcf.header)
+
+    for sv in vcf:
+        info = dict([*sv.INFO.items()])
+        svtype = info["SVTYPE"]
+        # only relevant for bnd-variants
+        if svtype != "BND":
+            writer.write_record(sv)
+            continue
+
+        # no need to rescue normal variants
+        if "TUMOR_PASS_CHROM" not in info:
+            writer.write_record(sv)
+            continue
+
+        bnd_id, sv_id_num = get_bnd_id(info)
+        sv.INFO["BND_ID"] = [bnd_id]
+
+        filter_bnds = bnd_filter_dict[bnd_id]
+        if "1" in filter_bnds and "2" in filter_bnds:
+            filter_status1 = filter_bnds["1"]
+            filter_status2 = filter_bnds["2"]
+            if "PASS" in filter_status1 or "PASS" in filter_status2:
+                sv.FILTER = ["PASS"]
+
+        writer.write_record(sv)
+
+
+@click.group()
+def main():
```

Comment:
Let's rename it to something more descriptive
```suggestion
def main():
```

Reply:
And also maybe move the CLI definition to the top 🔝

Reply:
It shall be done!